code/modules/atmospherics/machinery/components/binary_devices/volume_pump.dm

```diff
@@ -49,6 +55,10 @@
 	SSradio.remove_object(src,frequency)
 	return ..()
 
+/obj/machinery/atmospherics/components/binary/volume_pump/proc/set_on(active)
+	on = active
+	SEND_SIGNAL(src, COMSIG_VOLUME_PUMP_SET_ON, on)
+
```

Comment:
set_on is kinda a confusing name for this. Maybe set_activated?

Reply:
Also not sure how happy I am with creating a setting for the on var here, rather then at its base, Would that be too much work?

Reply:
I don't understand the ask here, it's so that circuits can tell when the volume pump is changed.

Reply:
They generally are, but there's specific and then there's redundant.
`COMSIG_PUMP_SET_ON` and `COMSIG_VOLUME_PUMP_SET_ON` will never be fired from the same object, because they're specific to the type of the atmos component being listened to. 

What we really want is a proc on /obj/machinery/atmospherics that sets the on variable, and sends a signal. That way anyone interested in an object's on variable changing has a easy way to listen to it, and we don't need to keep creating what amounts to copies of the same signal and proc for each thing we convert to circuits.

It's more upfront work is the thing, which is I assume why it hasn't been done in the past.

I feel like I'm doing a poor job of explaining this, sorry if it's not coming across very clearly.

Reply:
Nah, it makes sense. I'll try it out.